File: 004/Scala/src/main/scala/Fib.scala
/**
 * Created by naoki on 2015/06/29.
 */
object Fib{
  val phi = (1+Math.sqrt(5))/2

  def getFib(x:Int):Int = ((math.pow(phi,x)-math.pow(-phi,-x))/math.sqrt(5)).toInt

}

File: 004/Java/src/Fib.java
/**
 * Created by naoki on 2015/06/27.
 */
public class Fib {

    private static final double PHI = (1+Math.sqrt(5))/2;

    private Fib(){}

    public static int getFib(int x){
        if(x<=0)return 0;
        else return fibGT(x);
    }

    private static int fibGT(int x){
        return (int) ((Math.pow(getPhi(),x)-(Math.pow(-getPhi(),-x)))/Math.sqrt(5));
    }

    private static double getPhi(){
        return PHI;
    }

}
